source/codegen/common_helpers.py

```diff
@@ -83,9 +83,19 @@ def snake_to_pascal(snake_string):
 
 def pascal_to_camel(pascal_string):
   '''Returns a camelString for a given PascalString.'''
-  pascal_string = list(pascal_string)
-  pascal_string[0] = pascal_string[0].lower()
-  return ("".join(pascal_string))
+  # Full string all-caps: IEPE -> iepe
+  match = re.fullmatch(r"([A-Z]+)", pascal_string)
+  if match:
+    return match[1].lower()
+  
+  # Leading all-caps: IPAddress -> ipAddress
+  match = re.fullmatch(r"([A-Z]+)([A-Z].*)", pascal_string)
```

Comment:
`.*` handles numerics here, but it won't handle if it comes after uppercase acronym in the beginning.

e.g.
`IP2` - probably should be handled by first case above
`IP2Host`

I am not sure if we need to handle all of the cases now or handle them if we notice any such names during reviews in future (could be missed during review though when you are migrating big chunk of APIs).

Reply:
Good catch.